### PR TITLE
Add view model factory to demo

### PR DESCRIPTION
**Pull request by andyb129**
This related to issue #6 

This needs to be merged after PR #11 as branch wouldn't compile against master.

### CODE REVIEW

File: app/src/main/java/work/beltran/conductorviewmodel/MyViewModelController.kt [hunk 1]
```diff
@@ -2,19 +2,29 @@ package work.beltran.conductorviewmodel
...
 import kotlinx.android.synthetic.main.controller_viewmodel.view.*
 
-class MyViewModelController : ViewModelController() {
+class MyViewModelController() : ViewModelController() {
+
+    private lateinit var fragmentActivity: FragmentActivity
+
+    constructor(fragmentActivity: FragmentActivity) : this() {
+        this.fragmentActivity = fragmentActivity
+    }
```
Comment (miquelbeltran) [inline app/src/main/java/work/beltran/conductorviewmodel/MyViewModelController.kt]: You shouldn't be passing a fragment to a Controller as constructor parameter

**Comment by miquelbeltran** [comment]
Thanks for the intention, but the proposed solution is not correct.

It should be something like:

```
val viewModel = viewModelProvider(MyViewModelFactory()).get(MyViewModel::class.java)
```

File: app/src/main/java/work/beltran/conductorviewmodel/MyViewModelController.kt (hunk 1 above)
Comment (andyb129) [reply app/src/main/java/work/beltran/conductorviewmodel/MyViewModelController.kt]: OK, I have refactored this and removed constructor, but not 100% sure its right, so let me know if not. Thanks.

**Comment by miquelbeltran** [comment]
Sorry, I have doubts about merging this one. I don't want to replace the original demo code. I will probably do some changes but keep your original commits.

As you are doing hackoberfest, this will count to your contributions still, so no worries. Remember that any PR that is not marked as "Invalid" counts.

**Pull request closed without merging**